plugin/evm/validators_cache_test.go

```diff
@@ -0,0 +1,87 @@
+package evm
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/flare-foundation/flare/ids"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testData map[uint64]map[ids.ShortID]uint64
+
+func (testData) ByEpoch(uint64) (map[ids.ShortID]uint64, error) {
+	return map[ids.ShortID]uint64{}, nil
+}
+
+func randID() ids.ShortID {
+	id := ids.ShortID{}
+	rand.Read(id[:])
+
+	return id
+}
+
+func TestWithCacheSize(t *testing.T) {
+	wantCacheSize := uint(42)
+
+	cfg := CacheConfig{}
+	WithCacheSize(wantCacheSize)(&cfg)
+
+	assert.Equal(t, wantCacheSize, cfg.CacheSize)
+}
+
+func TestValidatorsCache_ByEpoch(t *testing.T) {
+	var (
+		idsList []ids.ShortID
+		weights []uint64
+
+		cacheTestData = testData{}
+	)
+
+	numValidators := 6
+	var epochs = [2]uint64{0, 1}
+	for i := 0; i < numValidators; i++ {
+		idsList = append(idsList, randID())
+		weights = append(weights, rand.Uint64())
+
+		if i >= numValidators/len(epochs) {
+			cacheTestData[epochs[0]] = map[ids.ShortID]uint64{
+				idsList[i]: weights[i],
+			}
+			continue
+		}
+
+		cacheTestData[epochs[1]] = map[ids.ShortID]uint64{
+			idsList[i]: weights[i],
+		}
+	}
+
+	validatorsCache := NewValidatorsCache(cacheTestData, WithCacheSize(32))
+	for k, v := range cacheTestData {
+		validatorsCache.cache.Add(k, v)
+	}
+
+	t.Run("NonExistingKey", func(t *testing.T) {
+		res, err := validatorsCache.ByEpoch(uint64(7))
+		require.NoError(t, err)
+		assert.Len(t, res, 0)
+	})
+
+	t.Run("ExistingKey", func(t *testing.T) {
+		for i, epoch := range epochs {
+			pos := numValidators - 1
+			if i == 1 {
+				pos = (numValidators / 2) - 1
+			}
+			result, err := validatorsCache.ByEpoch(epoch)
+			require.NoError(t, err)
+
+			value, ok := result[idsList[pos]]
+			assert.True(t, ok)
+
+			expected := cacheTestData[epoch][idsList[pos]]
+			assert.Equal(t, value, expected)
+		}
+	})
```

Comment:
This test should be defined first since it is the nominal case. Also be careful, `assert` functions take the expected value first and the assessed second

```suggestion
	t.Run("nominal case", func(t *testing.T) {
		for i, epoch := range epochs {
			result, err := validatorsCache.ByEpoch(epoch)
			require.NoError(t, err)

			pos := numValidators - 1
			if i == 1 {
				pos = (numValidators / 2) - 1
			}
			value, ok := result[idsList[pos]]
			assert.True(t, ok)

			want:= cacheTestData[epoch][idsList[pos]]
			assert.Equal(t, want, value)
		}
	})
```

Reply:
Moved it to first place and fixed:
https://github.com/flare-foundation/coreth/pull/21/commits/795a19e455a7a1b0b6a427200d807bfb6ec6d3dc#diff-53d507a9b6fbf6d169f0b9686ad727450a293914f998f4a17c524fa84623c4d7R57

Reply:
Moved it to first place and fixed:
https://github.com/flare-foundation/coreth/pull/21/commits/795a19e455a7a1b0b6a427200d807bfb6ec6d3dc#diff-53d507a9b6fbf6d169f0b9686ad727450a293914f998f4a17c524fa84623c4d7R57